src/main/java/com/epam/preprod/sirenko/command/impl/CreateNewProductCommand.java

```diff
@@ -34,19 +30,8 @@ public void execute() {
 				PrintToConsole.printString(INCORRECT_INPUT_CATEGORY_NAME);
 				return;
 			}
-			PrintToConsole.printString(CHOOSE_STRATEGY);
-			String inputStrategy = ConsoleReader.readFromConsole();
-			if (!ValidatorOfConsoleInput.checkInputStringIsNumberOneOrZero(inputStrategy)) {
-				PrintToConsole.printString(INCORRECT_INPUT_ONE_OR_ZERO);
-				inputStrategy = ConsoleReader.readFromConsole();
-			}
-			Strategy strategy;
-			if (inputStrategy.matches("[1]")) {
-				strategy = new CreatedByInputProductImpl();
-			} else {
-				strategy = new AutoGeneratedProductImpl();
-			}
-			productService.createNewProduct(strategy, inputCategory);
+			CreateProductFactory createProductFactory = factoryContainer.getFactory(inputCategory);
+			productService.createNewProduct(createProductFactory);
```

Comment:
As we have already discussed, you should not pass factory to service during service creation, but not in method for product entity creation.You can pass in in constructor or even pass factoryContainer to it

Reply:
I moved it to service but I still need factoryContainer here to print categories